src/commands/dev/dev.mjs

```diff
@@ -681,9 +695,92 @@ const dev = async (options, command) => {
   printBanner({ url })
 }
 
-const getBuildOptions = ({ cachedConfig, options: { context, cwd = process.cwd(), debug, dry, offline }, token }) => ({
+// Copies `netlify.toml`, if one is defined, into the `.netlify` internal
+// directory and returns the path to its new location.
+const copyConfig = async ({ configPath, siteRoot }) => {
+  const newConfigPath = path.resolve(siteRoot, getPathInProject(['netlify.toml']))
+
+  try {
+    await fs.copyFile(configPath, newConfigPath)
+  } catch {
+    // no-op
+  }
+
+  return newConfigPath
+}
+
+// Loads and runs Netlify Build. Chooses the right flags and entry point based
+// on the options supplied.
+const runBuild = async ({ cachedConfig, options, settings, site }) => {
+  const { default: buildSite, startDev } = await netlifyBuildPromise
+  const sharedOptions = getBuildOptions({
+    cachedConfig,
+    options,
+  })
+  const devCommand = async (settingsOverrides = {}) => {
+    const { ipVersion } = await startFrameworkServer({
+      settings: {
+        ...settings,
+        ...settingsOverrides,
+      },
+    })
+
+    settings.frameworkHost = ipVersion === 6 ? '::1' : '127.0.0.1'
+  }
+
+  if (options.prod) {
+    // Start by cleaning the internal directory, as it may have artifacts left
+    // by previous builds.
+    await cleanInternalDirectory(site.root)
+
+    // Copy `netlify.toml` into the internal directory. This will be the new
+    // location of the config file for the duration of the command.
+    const tempConfigPath = await copyConfig({ configPath: cachedConfig.configPath, siteRoot: site.root })
+    const buildSiteOptions = {
+      ...sharedOptions,
+      outputConfigPath: tempConfigPath,
+      saveConfig: true,
+    }
+
+    // Run Netlify Build using the main entry point.
+    await buildSite(buildSiteOptions)
```

Comment:
Agreed. I've done it in https://github.com/netlify/cli/pull/5351/commits/f4f31fb973e07d942454b625d34dd3022425081d to avoid merge conflicts.

Once you're happy with this PR, I can merge that one in.